daffodil-lib/src/main/scala/org/apache/daffodil/util/DataValue.scala

```diff
@@ -0,0 +1,91 @@
+package org.apache.daffodil.infoset
+
+import java.lang.{ Boolean => JBoolean, Number => JNumber, Long => JLong, Double => JDouble, String => JString, Float => JFloat, Byte => JByte, Integer => JInt, Short => JShort }
+import java.math.{ BigDecimal => JBigDecimal, BigInteger => JBigInt }
+import org.apache.daffodil.calendar.DFDLCalendar
+
+/*
+ * DINode is not visible to Daffodil-lib, so we cannot actually reference this here.
+ * We are trusting the rest of the codebase to only apply DataValueDINode to the DINode
+ * class.
+ * The DINode class itself defines an implicit DataValueDINode -> DINode conversion
```

Comment:
Why are you bothering to put this in daffodil-lib? It feels like part of the runtime to me, and this should just be part of InfosetImpl.scala or related file.

Reply:
I believe I was looking at our Numbers.scala file, which currently is defined in terms of AnyRef. I believe our only use of it is for objects that are DataValue; so I wanted to try include that in the code that is DataValue aware. Particuarly the as_ methods.

I suppose we could move these methods to DataValue itself. The issue there is that we need to distinguish between the accesors that just do an .asInstanceOf, and those that dispatch on the type and potentially do a runtime conversion.

I can try doing that, then hopefully remove most of what is in Numbers.scala.